### PR TITLE
[Merged by Bors] - feat(Data/Nat/Bitwise): add simp lemmas

### PR DESCRIPTION
**Pull request by pitmonticone**
Co-authored-by: @Command-Master 

Upstreamed from the [EquationalTheories](https://github.com/teorth/equational_theories) project.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Nat.lean
Comment (eric-wieser) [inline Mathlib/Algebra/Group/Nat.lean:1]: I think these maybe belong in `Nat/Bitwise`, unless the imports are tricky

Comment (pitmonticone) [reply Mathlib/Algebra/Group/Nat.lean:1]: Done. Thanks.

File: Mathlib/Data/Nat/Bitwise.lean [hunk 1]
```diff
@@ -354,6 +353,22 @@ theorem lt_xor_cases {a b c : ℕ} (h : a < b ^^^ c) : a ^^^ c < b ∨ a ^^^ b <
   obtain ha | hb | hc := xor_trichotomy <| Nat.xor_assoc _ _ _ ▸ xor_ne_zero.2 h.ne
   exacts [(h.asymm ha).elim, Or.inl <| Nat.xor_comm _ _ ▸ hb, Or.inr hc]
 
+@[simp]
+theorem xor_mod_two_eq {m n : ℕ} : (m ^^^ n) % 2 = (m + n) % 2 := by
+  by_cases h : (m + n) % 2 = 0
+  · simp only [h, mod_two_eq_zero_iff_testBit_zero, testBit_zero, xor_mod_two_eq_one, decide_not,
+    Bool.decide_iff_dist, Bool.not_eq_false', beq_iff_eq, decide_eq_decide]
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/Bitwise.lean:360]: Wrapped lines should be indented:
```suggestion
  · simp only [h, mod_two_eq_zero_iff_testBit_zero, testBit_zero, xor_mod_two_eq_one, decide_not,
     Bool.decide_iff_dist, Bool.not_eq_false', beq_iff_eq, decide_eq_decide]
```

File: Mathlib/Data/Nat/Bitwise.lean [hunk 2]
```diff
@@ -354,6 +353,22 @@ theorem lt_xor_cases {a b c : ℕ} (h : a < b ^^^ c) : a ^^^ c < b ∨ a ^^^ b <
...
+    omega
+  · simp only [mod_two_ne_zero] at h
+    simp only [h, xor_mod_two_eq_one]
+    omega
+
+@[simp]
+theorem even_xor {m n : ℕ} : Even (m ^^^ n) ↔ (Even m ↔ Even n) := by
+  simp only [even_iff, xor_mod_two_eq]
+  simp only [← even_iff]
```
Comment (eric-wieser) [inline Mathlib/Data/Nat/Bitwise.lean:369]: Perhap's you're lookign for
```suggestion
  simp_rw [even_iff, xor_mod_two_eq, ← even_iff]
```

Comment (pitmonticone) [reply Mathlib/Data/Nat/Bitwise.lean]: It returns an error `simp made no progress`, but I have modified it so that it's not necessary.

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**